src/components/member-role-update/index.js

```diff
@@ -26,48 +23,110 @@ const MemberRoleUpdate = () => {
     `${BASE_API_URL}/users/${selectedMember}`
   );
 
+  const member = userData?.user.roles.member;
+  const archived = userData?.user.roles.archived;
+  const userId = userData?.user.id;
   const { data: tagData } = useFetch(`${BASE_API_URL}/tags`);
   const { data: levelData } = useFetch(`${BASE_API_URL}/levels`);
 
-  const promoteToMember = async (user) => {
+  const promoteToMember = async (id) => {
     setIsUpdating(true);
-    const { status } = await moveToMember(user);
+    const role = {
+      member: true,
+    };
+    const { status } = await memberRoleUpdate(id, role);
     setIsUpdating(false);
-    if (status === 204) {
-      setUpdateStatus('User moved to member');
+    if (status === 200) {
+      setUpdateStatus('user promoted to a member');
+    } else {
+      setUpdateStatus('some error occured, please contact admin');
+    }
+  };
+
+  const demoteFromMember = async (id) => {
+    setIsUpdating(true);
+    const role = {
+      member: false,
+    };
+    const { status } = await memberRoleUpdate(id, role);
+    setIsUpdating(false);
+    if (status === 200) {
+      setUpdateStatus('user deomoted from member');
+    } else {
+      setUpdateStatus('some error occured, please contact admin');
+    }
+  };
+
+  const archiveTheMember = async (id) => {
+    setIsUpdating(true);
+    const role = {
+      archived: true,
+    };
+    const { status } = await memberRoleUpdate(id, role);
+    setIsUpdating(false);
+    if (status === 200) {
+      setUpdateStatus('user archived!');
     } else {
       setUpdateStatus('Some error occured, please contact admin');
     }
   };
```

Comment:
can we not create only two functions that will just toggle these roles, instead of having four function, I think that is possible